sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -436,7 +436,7 @@ class DataFrameStatSuite extends QueryTest with SharedSparkSession {
     val df = spark.range(0, 100)
       .select(lit("Foo").as("name"), (col("id") % 3).as("key"))
     val sampled = df.stat.sampleBy(
-      struct($"name", $"key"), Map(Row("Foo", 0) -> 0.1, Row("Foo", 1) -> 0.2), 0L)
+      struct($"name", $"key"), Map(Row("Foo", 0L) -> 0.1, Row("Foo", 1L) -> 0.2), 0L)
```

Comment:
what happens if we don't add the suffix `L`? If it fails when it seems a breaking change.

Reply:
Yes, this where I realized that changeing to `.equals` is probably not a good idea: https://github.com/apache/spark/pull/31682#issuecomment-787517417